internal/manifest/manifest.go

```diff
@@ -140,40 +147,94 @@ func (m *Manifest) Validate() error {
 	return nil
 }
 
-// AKSValidateOpts returns validate options populated with the manifest's
-// AKS reference values and trusted measurement.
-func (m *Manifest) AKSValidateOpts() (*validate.Options, error) {
-	if m.ReferenceValues.AKS == nil {
-		return nil, fmt.Errorf("no AKS reference values present in manifest")
+// SNPValidateOptsGenerator generates SNP validation options and
+// can be instantiated from a manifest only.
+type SNPValidateOptsGenerator struct {
+	opts        *validate.Options
+	manifest    *Manifest
+	extraChecks []func(report *sevsnp.Report) error // additional checks that need to pass for the validation to succeed.
+}
+
+// SNPValidateOpts returns the SNP validation options.
+func (g *SNPValidateOptsGenerator) SNPValidateOpts(report *sevsnp.Report) (*validate.Options, error) {
+	for _, check := range g.extraChecks {
+		if err := check(report); err != nil {
+			return nil, fmt.Errorf("additional check failed: %w", err)
+		}
+	}
+	return g.opts, nil
+}
+
+// TODO(msanft): add generic validation interface for other attestation types.
+
+// SNPValidateOpts returns validate options generators populated with the manifest's
+// SNP reference values and trusted measurement for the given runtime.
+func (m *Manifest) SNPValidateOpts() ([]*SNPValidateOptsGenerator, error) {
+	if m.ReferenceValues == nil {
+		return nil, errors.New("reference values cannot be empty")
 	}
 
 	if err := m.Validate(); err != nil {
 		return nil, fmt.Errorf("validating manifest: %w", err)
 	}
-	trustedMeasurement, err := m.ReferenceValues.AKS.TrustedMeasurement.Bytes()
-	if err != nil {
-		return nil, fmt.Errorf("failed to convert TrustedMeasurement from manifest to byte slices: %w", err)
-	}
-
-	return &validate.Options{
-		Measurement: trustedMeasurement,
-		GuestPolicy: abi.SnpPolicy{
-			Debug: false,
-			SMT:   true,
-		},
-		VMPL: new(int), // VMPL0
-		MinimumTCB: kds.TCBParts{
-			BlSpl:    m.ReferenceValues.AKS.SNP.MinimumTCB.BootloaderVersion.UInt8(),
-			TeeSpl:   m.ReferenceValues.AKS.SNP.MinimumTCB.TEEVersion.UInt8(),
-			SnpSpl:   m.ReferenceValues.AKS.SNP.MinimumTCB.SNPVersion.UInt8(),
-			UcodeSpl: m.ReferenceValues.AKS.SNP.MinimumTCB.MicrocodeVersion.UInt8(),
-		},
-		MinimumLaunchTCB: kds.TCBParts{
-			BlSpl:    m.ReferenceValues.AKS.SNP.MinimumTCB.BootloaderVersion.UInt8(),
-			TeeSpl:   m.ReferenceValues.AKS.SNP.MinimumTCB.TEEVersion.UInt8(),
-			SnpSpl:   m.ReferenceValues.AKS.SNP.MinimumTCB.SNPVersion.UInt8(),
-			UcodeSpl: m.ReferenceValues.AKS.SNP.MinimumTCB.MicrocodeVersion.UInt8(),
-		},
-		PermitProvisionalFirmware: true,
-	}, nil
+
+	var out []*SNPValidateOptsGenerator
+	for _, refVal := range m.ReferenceValues {
+		trustedMeasurement, err := refVal.SNP.TrustedMeasurement.Bytes()
```

Comment:
Don't we need to do a null-pointer check here?

Reply:
We could certainly check that here, but since `Validate` checks that already, I'm not sure whether it's necessary.

Reply:
No, validate only checks that either `SNP` **or** `TDX` is set. This code just assumes that `SNP` is set.

Reply:
> I refer to the SNP-specific `Validate`:
> 
> https://github.com/edgelesssys/contrast/blob/e604fdaae9b04d2c9e50abb55e6c8a9d51841c50/internal/manifest/manifest.go#L103-L105

This code isn't called when `SNP` is nil.

> > This code just assumes that SNP is set.
> 
> For this, see my comment regarding TDX availability below.

Sure, but IMHO crashing because of a null pointer dereference is bad even if a user had to manually change the manifest in order to trigger it. If we yet don't want users to not specify values for SNP then we should adjust `ReferenceValues.Validate` accordingly.